public/app/features/logs/logsModel.ts

```diff
@@ -789,3 +790,24 @@ function getIntervalInfo(scopedVars: ScopedVars, timespanMs: number): { interval
     return { interval: '$__interval' };
   }
 }
+
+/**
+ * Converts log row into data frame with a single row.
+ *
+ * @export
+ * @param {LogRowModel} logRow
+ * @return {DataFrame}
+ */
+export function logRowToDataFrame(logRow: LogRowModel): DataFrame | null {
+  const originFrame = logRow.dataFrame;
+
+  if (originFrame.length === 0 || originFrame.length <= logRow.rowIndex) {
+    return null;
+  }
+
+  const frame = createDataFrame({
```

Comment:
If `logRow` has a `DataFrame`, why can't you just return it?

Reply:
The crucial part is the `[field.values[logRow.rowIndex]]` that creates a new dataframe with just one row, based on `logRow.rowIndex`. I'll add a comment to make this clear.